### PR TITLE
Add programs to VET TEC Profile Page bah-19614

### PR DESCRIPTION
**Pull request by hughes-dustin**
## Description
https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/vets.gov-team/19614

## Acceptance Criteria
- [ ] If the provider has no programs, the following text is displayed: "Program data will be available for this provider soon. Learn more about VET TEC programs"
     a. "Learn more about VET TEC programs" is a link to the following page: https://staging.va.gov/education/about-gi-bill-benefits/how-to-use-benefits/vettec-high-tech-program/
- [ ]  If the provider has programs, the following is displayed:
     a. "Select a program below to view your estimated  benefits."
     b. A Table with the following columns: Program name, Length
     c. A row per program that displays the information loaded via the "Program" file in GIDS.

### CODE REVIEW

File: app/models/institution.rb [hunk 1]
```diff
@@ -210,6 +210,10 @@ def school?
     institution_type_name != 'OJT'
   end
 
+  def programs
+    Program.where('facility_code' => self.facility_code)
```
Comment (shawkey-daniel) [inline app/models/institution.rb]: Code climate says self is redundant.

**Review by shawkey-daniel — CHANGES_REQUESTED** [review]

File: app/models/institution.rb (hunk 1 above)
Comment (hughes-dustin) [reply app/models/institution.rb]: Changes are in. Github just took it's time showing the update.

**Review by shawkey-daniel — APPROVED** [review]
LGTM

File: app/models/institution.rb [hunk 2]
```diff
@@ -210,6 +210,10 @@ def school?
     institution_type_name != 'OJT'
   end
 
+  def programs
+    Program.where('facility_code' => facility_code)
```
Comment (shawkey-daniel) [inline app/models/institution.rb]: Matt asked Joe to use https://guides.rubyonrails.org/association_basics.html#has-many-association-reference. We should be consistent and use that here also.

**Review by shawkey-daniel — CHANGES_REQUESTED** [review]

File: app/models/institution.rb (hunk 2 above)
Comment (hughes-dustin) [reply app/models/institution.rb]: Added in the changes.

**Review by shawkey-daniel — CHANGES_REQUESTED** [review]
The has_many association should have a dependent: destroy for cascade delete and you may want a belongs_to in program.

File: app/models/program.rb [hunk 3]
```diff
@@ -15,4 +15,6 @@ class Program < ActiveRecord::Base
   }.freeze
 
   validates :facility_code, presence: true
+  belongs_to :insitutions
```
Comment (shawkey-daniel) [inline app/models/program.rb]: Code climate detected trailing whitespace! Unacceptable!

**Review by shawkey-daniel — DISMISSED** [review]

**Review by zurbergram — APPROVED** [review]
LGTM

**Review by delli-gatti-michael — APPROVED** [review]
Approved by QA

File: app/models/program.rb [hunk 4]
```diff
@@ -15,4 +15,5 @@ class Program < ActiveRecord::Base
...
```
Comment (kfrz) [inline app/models/program.rb]: This should be `belongs_to :institution` I think

**Review by kfrz — DISMISSED** [review]
Overall looks good. I think with the long serializer it may be useful to extract out commonly used attributes, like address, out to a concern, which would reduce the length of the class as well as make the code more DRY overall.

File: app/models/program.rb (hunk 4 above)
Comment (hughes-dustin) [reply app/models/program.rb]: Nice catch. That definitely was a typo.

File: app/models/program.rb [hunk 5]
```diff
@@ -15,4 +15,5 @@ class Program < ActiveRecord::Base
   }.freeze
 
   validates :facility_code, presence: true
+  belongs_to :institution
```
Comment (jholton) [inline app/models/program.rb]: This line should probably just be deleted.
`programs` doesn't have a `institution_id` field.
It does have a `facility_code`, but it doesn't have a `version` attribute.  That is, there will be multiple `institutions` records (one per version) for each `program`

File: app/serializers/program_serializer.rb [hunk 6]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+class ProgramSerializer < ActiveModel::Serializer
+  attribute :facility_code
+  attribute :institution_name
```
Comment (jholton) [inline app/serializers/program_serializer.rb]: The two above attributes should probably be removed.  It's redundant data that is available on the `institutions` object.

File: app/models/program.rb (hunk 5 above)
Comment (hughes-dustin) [reply app/models/program.rb]: Got it. Changes have been submitted.

File: app/serializers/program_serializer.rb (hunk 6 above)
Comment (hughes-dustin) [reply app/serializers/program_serializer.rb]: That's a 10-4. Changes are in.

**Review by jholton — APPROVED** [review]
lgtm

**Pull request merged**